app/lib/common/widgets/acter_search_widget.dart

```diff
@@ -1,48 +1,84 @@
 import 'package:atlas_icons/atlas_icons.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
-import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-final searchValueProvider = StateProvider.autoDispose<String>((ref) => '');
+class ActerSearchWidget extends StatefulWidget {
```

Comment:
Optimised `ActerSearchWidget` code to make it generic in nature and can be used across entire app based on all the use-case requirements.